From 0826974c5a91ba09bebf513fe4000dc11af6bc52 Mon Sep 17 00:00:00 2001
From: Pete Vander Giessen <petevg@gmail.com>
Date: Tue, 19 Jul 2016 16:40:35 -0400
Subject: [PATCH] BIGTOP-2504: Kafka should be able to bind to something other
 than 0.0.0.0/the default interface

If set to something that ruby evaluates as true, will be written to
server.properties. Causes the server to listen on the ip/interface
indicated by the value of the string.
---
 bigtop-deploy/puppet/modules/kafka/manifests/init.pp           | 3 ++-
 bigtop-deploy/puppet/modules/kafka/templates/server.properties | 4 ++++
 2 files changed, 6 insertions(+), 1 deletion(-)

diff --git a/bigtop-deploy/puppet/modules/kafka/manifests/init.pp b/bigtop-deploy/puppet/modules/kafka/manifests/init.pp
index f44b737..3394481 100644
--- a/bigtop-deploy/puppet/modules/kafka/manifests/init.pp
+++ b/bigtop-deploy/puppet/modules/kafka/manifests/init.pp
@@ -24,7 +24,8 @@
   class server(
       $broker_id = "0",
       $port = "9092",
-      $zookeeper_connection_string = "localhost:2181"
+      $zookeeper_connection_string = "localhost:2181",
+      $bind_addr = ""
     ) {
 
     package { 'kafka':
diff --git a/bigtop-deploy/puppet/modules/kafka/templates/server.properties b/bigtop-deploy/puppet/modules/kafka/templates/server.properties
index a30e970..6e85413 100644
--- a/bigtop-deploy/puppet/modules/kafka/templates/server.properties
+++ b/bigtop-deploy/puppet/modules/kafka/templates/server.properties
@@ -25,7 +25,11 @@ broker.id=<%= @broker_id %>
 port=<%= @port %>
 
 # Hostname the broker will bind to. If not set, the server will bind to all interfaces
+<% if !@bind_addr.nil? && !@bind_addr.empty? %>
+host.name=<%= @bind_addr %>
+<% else %>
 #host.name=localhost
+<% end %>
 
 # Hostname the broker will advertise to producers and consumers. If not set, it uses the
 # value for "host.name" if configured.  Otherwise, it will use the value returned from
